stanford_syndication.module

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for stanford_syndication module.
+ */
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\node\NodeInterface;
+use Drupal\stanford_syndication\Events\SyndicationEntityActionEventInterface;
+use Drupal\stanford_syndication\Events\SyndicationEntityActionEvent;
+use Drupal\stanford_syndication\Events\SyndicationEvents;
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function stanford_syndication_node_insert(NodeInterface $node) {
+  drupal_register_shutdown_function('_stanford_syndication_dispatch_event', $node, SyndicationEntityActionEventInterface::INSERT_ACTION);
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function stanford_syndication_node_update(NodeInterface $node) {
+  drupal_register_shutdown_function('_stanford_syndication_dispatch_event', $node, SyndicationEntityActionEventInterface::UPDATE_ACTION);
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_delete().
+ */
+function stanford_syndication_node_delete(NodeInterface $node) {
+  drupal_register_shutdown_function('_stanford_syndication_dispatch_event', $node, SyndicationEntityActionEventInterface::DELETE_ACTION);
+}
+
+/**
+ * Dispatches the syndication event.
+ *
+ * @param \Drupal\stanford_syndication\Events\SyndicationEntityActionEventInterface $event
+ *   Event to dispatch.
+ */
+function _stanford_syndication_dispatch_event(NodeInterface $node, string $action) {
+  if ($node->get('syndication')->getString() && $node->isPublished()) {
+    $event = new SyndicationEntityActionEvent($node, $action);
+    \Drupal::service('event_dispatcher')
+      ->dispatch($event, SyndicationEvents::ENTITY_ACTION);
+  }
+}
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function stanford_syndication_entity_base_field_info(EntityTypeInterface $entity_type) {
+  $fields = [];
+  if ($entity_type->id() == 'node') {
+    $fields['syndication'] = BaseFieldDefinition::create('boolean')
+      ->setLabel(t('Syndicate Content'))
+      ->setDescription(t('Send a notification to the appropriate systems about this piece of content.'))
+      ->setStorageRequired(TRUE)
```

Comment:
Don't quote me on this (I have to pull down and test first) but I believe requiring a boolean/checkbox field means you can't uncheck it.

Reply:
no, this just adds the storage into the `node_field_data` table. By default the checkbox is unchecked at the moment.

Reply:
it's not saying the field is required, just that storage is needed for the base field.

Reply:
Oooooooh. Gotcha 🤦